jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ResponseTest.java

```diff
@@ -56,6 +59,97 @@ public void dispose() throws Exception
         connector = null;
     }
 
+    @Test
+    public void testGET() throws Exception
+    {
+        server.setHandler(new Handler.Abstract()
+        {
+            @Override
+            public boolean handle(Request request, Response response, Callback callback)
+            {
+                callback.succeeded();
+                return true;
+            }
+        });
+        server.start();
+
+        String request = """
+                GET /path HTTP/1.0\r
+                Host: hostname\r
+                \r
+                """;
+        HttpTester.Response response = HttpTester.parseResponse(connector.getResponse(request));
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+    }
+
+    @Test
+    public void testServerDateFieldsFrozen() throws Exception
+    {
+        server.setHandler(new Handler.Abstract()
+        {
+            @Override
+            public boolean handle(Request request, Response response, Callback callback)
```

Comment:
If an assertion fails within the handler, the callback won't be succeeded and the test will need to wait until the timeout before finishing and be marked as failed. IMHO at least the timeout should be reduced to a few seconds.